src/frontend/src/scheduler/distributed/query_manager.rs

```diff
@@ -176,7 +181,7 @@ impl QueryResultFetcher {
             .await?;
         let mut stream = compute_client.get_data(self.task_output_id.clone()).await?;
         while let Some(response) = stream.next().await {
-            yield DataChunk::from_protobuf(response.to_rw_result()?.get_record_batch()?)?
+            yield DataChunk::from_protobuf(response.map_err(TonicStatus)?.get_record_batch()?)?;
```

Comment:
We may also use an `RpcError` variant for this and implement `From<..>` for this so there's no need to `map_err` here.

Reply:
`yield` returns `RwError`, tonic::Status -> SchedulerError::RpcError -> RwError, at least one `map` is needed I guess?